src/btree/bt_curnext.c

```diff
@@ -900,13 +914,18 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
          * from reconciliation getting rid of the obsolete content. Hence mark the page dirty to
          * force it through reconciliation.
          */
-        if (page != NULL &&
-          (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD ||
-            (newpage && cbt->page_deleted_count > 0))) {
-            WT_ERR(__wt_page_dirty_and_evict_soon(session, cbt->ref));
-            WT_STAT_CONN_INCR(session, cache_eviction_force_delete);
+        if (page != NULL) {
+            if (cbt->page_obsolete_deleted_count > WT_BTREE_DELETE_THRESHOLD ||
+              (newpage && cbt->page_obsolete_deleted_count > 0)) {
+                WT_ERR(__wt_page_dirty_and_evict_soon(session, cbt->ref));
+                WT_STAT_CONN_INCR(session, cache_eviction_force_obsolete_delete);
+            } else if (page->type != WT_PAGE_COL_FIX && total_skipped != 0 && !cbt->valid_data) {
```

Comment:
Can you write a comment and explain what is being done here. I am finding it hard to read.

Reply:
The visible tombstone is either treated as obsolete delete or a page with non-deleted pages. I added a comment for both, please check.